Optimized the options and implemented the reserved_case option

Shun7 opened the pull request:
Optimized the structure of the options and implemented the reserved_case option to be able to change case of reserved words.

Commit:
予約語のみ大文字に変換するオプションを追加。
▽変更内容
　・README（英語、日本語）
　・プログラム本体（init.py, api.py, config.py, filters.py）
　・Sublime向け対応としてPreferences.sublime-settingsを改修

Commit:
オプション構成の最適化を実施。

Commit:
コード修正に伴い、Readmeも修正。

Commit:
Readmeを修正。

Commit:
sublimeプラグイン実行時に不具合があったので修正。

Shun7 commented:
Tested on my local pc and found some bugs. Now it works fine. So please check the code and pull the code!

Commit:
Change the default settings of the reserved words option to ANSI SQL(SQL:2011).

Commit:
Added the 'nochange' option.

Commit:
Updated the readme files based on the functional update which I made before.

Commit:
Deleted the blank line.

Shun7 commented:
@ota-meshi 
Added the case, reserved case and reserved word options instead of just the uppercase option. Also updated the readme files. Could you please check the code and pull the request if everything is ok.

ota-meshi commented on urobosql_formatter.py:
```diff
@@ -24,13 +24,25 @@ def run(self, edit, sync=False):
+        if self.getval("uf_case") == 'nochange':
+            config.set_case(None)
+        else:
+            config.set_case(self.getval("uf_case"))
```
For backward compatibility if 
```py
self.user_settings.get("uf_uppercase") == False and self.user_settings.get("uf_upper") == None
```
, please behavior as 
```py
config.set_case(None)
```

ota-meshi replied:
The same applies to `"uf_reserved_case"`

Shun7 committed:
Changed the default setting of the "uf_case". The default will be "upper"

Shun7 committed:
Added a code to manage backward compatibility.

Shun7 commented:
@ota-meshi 
Modified the code to manage backward compatibility. Please review the code again.

shout-star committed:
- fixed the problem with the execution timing of backward compatible code
- fixed the problem that the `uf_reserved_words` setting refers only to the user setting

shout-star committed:
update readme

ota-meshi commented:
close #4
